android/app/src/main/java/org/fossasia/openevent/viewmodels/TrackSessionsActivityViewModel.java

```diff
@@ -0,0 +1,126 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.dbutils.FilterableRealmLiveData;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.utils.DateConverter;
+import org.fossasia.openevent.utils.DateService;
+import org.threeten.bp.ZonedDateTime;
+
+import java.util.List;
+import java.util.Locale;
+
+import io.reactivex.functions.Predicate;
+import io.realm.RealmChangeListener;
+import io.realm.RealmResults;
+
+public class TrackSessionsActivityViewModel extends ViewModel {
+
+    private RealmDataRepository realmRepo;
+    private Track track;
+    private LiveData<Track> trackLiveData;
+    private FilterableRealmLiveData<Session> filteredSessionsLiveData;
+    private RealmResults<Session> sessions;
+    private String searchText = "";
+    private int ongoingPosition;
+    private int upcomingPosition;
+    private int flag;
+    private boolean firstTime = true;
+
+    public TrackSessionsActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<Track> getTrack(int trackId) {
+        if (trackLiveData == null) {
+            track = realmRepo.getTrack(trackId);
+            track.addChangeListener((realmModel, changeSet) -> {
+                sessions = track.getSessions().sort("startsAt");
+                if (filteredSessionsLiveData == null) {
+                    filteredSessionsLiveData = new FilterableRealmLiveData<>(sessions);
+                }
+                sessions.addChangeListener(this::calculateUpcomingOngoingPosition);
+            });
+            trackLiveData = RealmDataRepository.asLiveDataForObject(track);
+        }
+        return trackLiveData;
+    }
+
+    private void filterData(String searchText) {
+        if (!this.searchText.equals(searchText) || firstTime) {
+            firstTime = false;
```

Comment:
Why we need firstTime

Reply:
Yes, we don't need it. I was getting a blank screen so i tried to avoid it. I found the actual bug now, it was actually in my implementation of the FilterableRealmLiveData.